cpp/Tuparg.hpp

```diff
@@ -10,7 +10,7 @@ namespace hsd {
         void* addr;
 
         template <typename X>
-        X& getnext() {
+        X& getnext_unsafe() {
             assert(addr);
```

Comment:
What happens if `addr` is nullptr that you really need to use an `assert`?

Reply:
It's more of a safety measure. The program normally shouldn't allow this function to be called with null pointer, but if it gets called (by some other way than `getnext`), then it's a debugging check, and it should be erased when compiling for release